### PR TITLE
[TIMOB-25492] Remove assert / file references

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-25492

Also uses:
- https://github.com/appcelerator/APSHTTPClient/pull/42
- https://github.com/appcelerator/aps_sdk/pull/303

### CODE REVIEW

File: iphone/iphone/Titanium.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -3002,6 +3003,7 @@
 					"\"$(SRCROOT)/../Classes/APSAnalytics\"",
 					"\"$(SRCROOT)/../Classes/APSHTTPClient\"",
 				);
+				OTHER_CFLAGS = "-DNDEBUG";
```
Comment (hansemannn) [inline iphone/iphone/Titanium.xcodeproj/project.pbxproj]: @cb1kenobi Question here: Can I do it this way or should I add it via the _build.js? I am not sure if our vanilla project may strip these settings.

File: iphone/Classes/TiBase.h [hunk 2]
```diff
@@ -70,6 +70,11 @@ extern "C" {
   {                            \
     TiLogMessage(__VA_ARGS__); \
   }
+  
+// Hide assertions in prod-builds
+#ifdef NDEBUG
+#define assert(e)   ((void)0)
```
Comment (janvennemann) [inline iphone/Classes/TiBase.h]: Shouldn't assertions automatically be disabled by the compiler if you pass the `-DNDEBUG` flag?

Comment (hansemannn) [reply iphone/Classes/TiBase.h]: Just tried it out, looks okay! PR updated.

File: iphone/Classes/TiBase.h [hunk 3]
```diff
@@ -118,8 +118,12 @@ NSString *JavascriptNameForClass(Class c);
     x = [y retain];               \
   }
 
+#ifdef DEBUG
 #define CODELOCATION [NSString stringWithFormat:@"%s (%@:%d)", __FUNCTION__, [[NSString stringWithFormat:@"%s", __FILE__] lastPathComponent], __LINE__]
-
+#else
+#define CODELOCATION @""
+#endif
+
```
Comment (janvennemann) [inline iphone/Classes/TiBase.h]: You have some superfluous whitespaces here which cause linting to fail.

**Review by janvennemann — CHANGES_REQUESTED** [review]

**Comment by hansemannn** [comment]
@janvennemann Updated.

**Review by janvennemann — APPROVED** [review]
Looks good, approved!

@hansemannn Both dependent PRs are merged. Does this one need an update or can it already be merged?

**Pull request merged**